src/parser.rs

```diff
@@ -396,10 +392,13 @@ impl Parser<'_> {
         } else if keyword == self.print_keyword {
             let mut values = vec![self.parse_expression(current_query)?];
             self.consume_whitespace();
-            while self.try_next() == Some(',') {
+            while self.peek()? == ',' {
```

Comment:
Ah, I was just about to bring this up myself (having seen how multiple attributes are parsed in the meantime). 

I think ultimately an error in `peek()` will be fatal anyway, since that can only happen when there is an unexpected EOF, and since `print` statements can only appear inside stanzas, they cannot appear immediately before EOF.

However, from that point of view, the parse error is _not_ with the `print` statement itself, and so it makes more sense to build the value and make the error appear when we try to close the surrounding stanza.